app/presenters/explorer_presenter.rb

```diff
@@ -122,166 +127,88 @@ def [](key)
-      @out << set_element_visible(el, visible)
-    end
-
-    # Scroll to top of main div
-    @out << "$('#main_div').scrollTop(0);"
-
-    @out << "$('h1#explorer_title > span#explorer_title_text').html('#{j ERB::Util.h(URI.unescape(@options[:right_cell_text]))}');" if @options[:right_cell_text]
-
-    # Reload toolbars
-    @options[:reload_toolbars].each_pair do |div_name, toolbar|
-      # we need to render even empty toolbar to actually remove the buttons
-      # that might be there
-      @out << javascript_pf_toolbar_reload("#{div_name}_tb", Array(toolbar)).html_safe
-    end
-
-    # reset miq_record_id, else it remembers prev id and sends it when add is pressed from list view
-    [:record_id, :parent_id, :parent_class].each { |variable| @out << set_or_undef(variable) }
-
-    # Open, select, and focus node in current tree
-    @out << "miqDynatreeActivateNodeSilently('#{@options[:active_tree]}', '#{@options[:osf_node]}');" unless @options[:osf_node].blank?
-
-    @options[:lock_unlock_trees].each { |tree, lock| @out << tree_lock(tree, lock) }
-
-    if @options[:load_chart]
-      @out << 'ManageIQ.charts.chartData = ' + @options[:load_chart].to_json + ';'
-      @out << Charting.js_load_statement(true)
-    end
-
-    @out << 'ManageIQ.changes = null;'                       if @options[:reset_changes]
-    @out << 'ManageIQ.oneTransition.oneTrans = 0;'           if @options[:reset_one_trans]
-    @out << 'ManageIQ.oneTransition.IEButtonPressed = true;' if @options[:one_trans_ie]
-
-    if @options[:focus]
-      @out << "if ($('##{@options[:focus]}').length) $('##{@options[:focus]}').focus();"
-    end
-
-    @out << "$('#clear_search').#{@options[:clear_search_show_or_hide]}();" if @options[:clear_search_show_or_hide]
-    # always replace content partial to adjust height of content div
-    @out << "miqInitMainContent();"
-    @out << "$('#quicksearchbox').modal('hide');" if @options[:hide_modal]
-
-    @out << "miqInitAccordions();" if @options[:init_accords]
-
-    # Don't turn off spinner for charts/timelines
-    @out << set_spinner_off unless @options[:ajax_action]
-  end
-
-  def build_calendar
-    if @options[:build_calendar].kind_of? Hash
-      calendar_options = @options[:build_calendar]
-    else
-      calendar_options = {}
-    end
-
-    js_build_calendar(calendar_options)
+    data
   end
 
-  # Fire an AJAX action
-  def ajax_action(options)
-    url = [options[:controller], options[:action], options[:record_id]].join('/')
-    "miqAsyncAjax('/#{url}');"
-  end
+  private
 
-  # Set a JS variable to value from options or 'undefined'
-  def set_or_undef(variable)
-    if @options[variable]
-      "ManageIQ.record.#{variable.to_s.camelize(:lower)} = '#{@options[variable]}';"
-    else
-      "ManageIQ.record.#{variable.to_s.camelize(:lower)} = null;"
+  def format_calendar_dates(options)
+    return {} unless @options[:build_calendar].kind_of?(Hash)
+    {}.tap do |h|
+      %i(date_from date_to).each do |key|
```

Comment:
Minor but I think `%i(date_from date_to).each_with_object({}) do |key, h|` would read much better than the `{}.tap...`